src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
   }
 
+  /**
+   * Gets primary amp core services.
+   * @return {!Performance}
+   */
+  init() {
+    this.viewer_ = viewerFor(this.win);
+    this.resources_ = resourcesFor(this.win);
+    return this;
+  }
+
+  /**
+   * Listens to viewer and resource events.
+   */
+  listen() {
+    if (!(this.viewer_ && this.resources_)) {
+      return;
+    }
+
+    // User requested document to be visible
+    this.viewer_.whenFirstVisible().then(() => {
+      this.isViewerVisible_ = true;
+      this.tick('rdc');
+    });
+
+    this.resources_.whenFirstResourceLaidOut().then(() => {
+      if (this.isViewerVisible_) {
+        // Prerender Completed, but user already
```

Comment:
from my understanding, this should be when a viewer requests a page to pre-render then the user has asked that page to be in view, but the pre-render hasn't completed yet

/cc @cramforce (correct me if i'm wrong)